SIL.WritingSystems/LdmlDataMapper.cs

```diff
@@ -806,7 +806,15 @@ public void Write(string filePath, WritingSystemDefinition ws, Stream oldFile)
 						DtdProcessing = DtdProcessing.Parse
 					};
 					reader = XmlReader.Create(oldFile, readerSettings);
-					element = XElement.Load(reader);
+					try
+					{
+						element = XElement.Load(reader);
+					}
+					catch
```

Comment:
I couldn't think of any exception I would want to handle differently. Things like OutOfMemory shouldn't be caught since with .Net 4.0 those errors are not catchable by default. The XmlException is the only exception I've seen so far, so that may be sufficient.